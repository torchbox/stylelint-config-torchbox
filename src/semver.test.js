const path = require('path');
const stylelint = require('stylelint');

const pkg = require('../package.json');

describe('semver - should those tests break, consider releasing a new major version of the package', () => {
    it('dependencies', () => {
        expect(pkg.dependencies).toMatchInlineSnapshot(`
            Object {
              "stylelint-a11y": "^1.2.1",
              "stylelint-config-prettier": "^5.2.0",
              "stylelint-config-standard": "^18.3.0",
              "stylelint-order": "^3.1.1",
              "stylelint-scss": "^3.10.0",
            }
        `);
    });

    it('peerDependencies', () => {
        expect(pkg.peerDependencies).toMatchInlineSnapshot(`
            Object {
              "stylelint": "^10.1.0",
            }
        `);
    });

    it('config contents', () => {
        const linter = stylelint.createLinter({
            configFile: path.join(__dirname, '..', 'config.js'),
        });
        return linter.getConfigForFile().then((result) => {
            expect(result.config.rules).toMatchSnapshot();
        });
    });
});
