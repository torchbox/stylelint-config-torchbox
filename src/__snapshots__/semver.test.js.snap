// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`semver - should those tests break, consider releasing a new major version of the package config contents 1`] = `
Object {
  "a11y/content-property-no-static-value": Array [
    true,
  ],
  "a11y/no-obsolete-attribute": Array [
    true,
  ],
  "a11y/no-obsolete-element": Array [
    true,
  ],
  "a11y/no-outline-none": Array [
    true,
  ],
  "a11y/no-text-align-justify": Array [
    true,
  ],
  "at-rule-empty-line-before": null,
  "at-rule-name-case": null,
  "at-rule-name-newline-after": null,
  "at-rule-name-space-after": Array [
    "always",
  ],
  "at-rule-no-unknown": null,
  "at-rule-no-vendor-prefix": Array [
    true,
  ],
  "at-rule-semicolon-newline-after": null,
  "at-rule-semicolon-space-before": Array [
    "never",
  ],
  "block-closing-brace-empty-line-before": null,
  "block-closing-brace-newline-after": null,
  "block-closing-brace-newline-before": null,
  "block-closing-brace-space-after": null,
  "block-closing-brace-space-before": null,
  "block-no-empty": Array [
    true,
  ],
  "block-opening-brace-newline-after": null,
  "block-opening-brace-newline-before": null,
  "block-opening-brace-space-after": null,
  "block-opening-brace-space-before": null,
  "color-hex-case": null,
  "color-hex-length": Array [
    "short",
  ],
  "color-named": Array [
    "never",
  ],
  "color-no-invalid-hex": Array [
    true,
  ],
  "comment-empty-line-before": Array [
    "always",
    Object {
      "except": Array [
        "first-nested",
      ],
      "ignore": Array [
        "stylelint-commands",
      ],
    },
  ],
  "comment-no-empty": Array [
    true,
  ],
  "comment-whitespace-inside": Array [
    "always",
  ],
  "custom-property-empty-line-before": Array [
    "always",
    Object {
      "except": Array [
        "after-custom-property",
        "first-nested",
      ],
      "ignore": Array [
        "after-comment",
        "inside-single-line-block",
      ],
    },
  ],
  "declaration-bang-space-after": Array [
    "never",
  ],
  "declaration-bang-space-before": Array [
    "always",
  ],
  "declaration-block-no-duplicate-properties": Array [
    true,
    Object {
      "ignore": Array [
        "consecutive-duplicates-with-different-values",
      ],
    },
  ],
  "declaration-block-no-shorthand-property-overrides": Array [
    true,
  ],
  "declaration-block-semicolon-newline-after": null,
  "declaration-block-semicolon-newline-before": null,
  "declaration-block-semicolon-space-after": null,
  "declaration-block-semicolon-space-before": null,
  "declaration-block-single-line-max-declarations": Array [
    1,
  ],
  "declaration-block-trailing-semicolon": null,
  "declaration-colon-newline-after": null,
  "declaration-colon-space-after": null,
  "declaration-colon-space-before": null,
  "declaration-empty-line-before": null,
  "declaration-no-important": Array [
    true,
  ],
  "font-family-no-duplicate-names": Array [
    true,
  ],
  "font-family-no-missing-generic-family-keyword": Array [
    true,
  ],
  "function-calc-no-invalid": Array [
    true,
  ],
  "function-calc-no-unspaced-operator": Array [
    true,
  ],
  "function-comma-newline-after": null,
  "function-comma-newline-before": null,
  "function-comma-space-after": null,
  "function-comma-space-before": null,
  "function-linear-gradient-no-nonstandard-direction": Array [
    true,
  ],
  "function-max-empty-lines": null,
  "function-name-case": Array [
    "lower",
  ],
  "function-parentheses-newline-inside": null,
  "function-parentheses-space-inside": null,
  "function-whitespace-after": Array [
    "always",
  ],
  "indentation": null,
  "keyframe-declaration-no-important": Array [
    true,
  ],
  "length-zero-no-unit": Array [
    true,
  ],
  "max-empty-lines": null,
  "max-line-length": null,
  "max-nesting-depth": Array [
    4,
  ],
  "media-feature-colon-space-after": Array [
    "always",
  ],
  "media-feature-colon-space-before": Array [
    "never",
  ],
  "media-feature-name-case": null,
  "media-feature-name-no-unknown": Array [
    true,
  ],
  "media-feature-name-no-vendor-prefix": Array [
    true,
  ],
  "media-feature-parentheses-space-inside": Array [
    "never",
  ],
  "media-feature-range-operator-space-after": Array [
    "always",
  ],
  "media-feature-range-operator-space-before": Array [
    "always",
  ],
  "media-query-list-comma-newline-after": null,
  "media-query-list-comma-newline-before": null,
  "media-query-list-comma-space-after": null,
  "media-query-list-comma-space-before": null,
  "no-descending-specificity": null,
  "no-duplicate-at-import-rules": Array [
    true,
  ],
  "no-duplicate-selectors": Array [
    true,
  ],
  "no-empty-source": Array [
    true,
  ],
  "no-eol-whitespace": null,
  "no-extra-semicolons": null,
  "no-invalid-double-slash-comments": Array [
    true,
  ],
  "no-missing-end-of-source-newline": null,
  "number-leading-zero": Array [
    "always",
  ],
  "number-no-trailing-zeros": null,
  "order/order": Array [
    Array [
      "dollar-variables",
      "custom-properties",
      Object {
        "hasBlock": false,
        "type": "at-rule",
      },
      "declarations",
    ],
  ],
  "property-case": null,
  "property-no-unknown": Array [
    true,
  ],
  "property-no-vendor-prefix": Array [
    true,
  ],
  "rule-empty-line-before": null,
  "scss/at-mixin-argumentless-call-parentheses": Array [
    "always",
  ],
  "scss/at-rule-no-unknown": Array [
    true,
  ],
  "scss/declaration-nested-properties": Array [
    "never",
  ],
  "scss/dollar-variable-no-missing-interpolation": Array [
    true,
  ],
  "scss/percent-placeholder-pattern": Array [
    "^do-not-use-placeholders$",
  ],
  "scss/selector-no-redundant-nesting-selector": Array [
    true,
  ],
  "selector-attribute-brackets-space-inside": Array [
    "never",
  ],
  "selector-attribute-operator-space-after": Array [
    "never",
  ],
  "selector-attribute-operator-space-before": Array [
    "never",
  ],
  "selector-attribute-quotes": Array [
    "always",
  ],
  "selector-combinator-space-after": null,
  "selector-combinator-space-before": null,
  "selector-descendant-combinator-no-non-space": null,
  "selector-list-comma-newline-after": null,
  "selector-list-comma-newline-before": null,
  "selector-list-comma-space-after": null,
  "selector-list-comma-space-before": null,
  "selector-max-empty-lines": null,
  "selector-max-id": Array [
    0,
  ],
  "selector-max-specificity": Array [
    "0,4,0",
  ],
  "selector-max-type": Array [
    2,
    Object {
      "ignore": Array [
        "child",
        "compounded",
        "next-sibling",
      ],
    },
  ],
  "selector-no-vendor-prefix": Array [
    true,
  ],
  "selector-pseudo-class-case": null,
  "selector-pseudo-class-no-unknown": Array [
    true,
  ],
  "selector-pseudo-class-parentheses-space-inside": Array [
    "never",
  ],
  "selector-pseudo-element-case": null,
  "selector-pseudo-element-colon-notation": Array [
    "double",
  ],
  "selector-pseudo-element-no-unknown": Array [
    true,
  ],
  "selector-type-case": Array [
    "lower",
  ],
  "selector-type-no-unknown": Array [
    true,
  ],
  "string-no-newline": Array [
    true,
  ],
  "string-quotes": null,
  "unit-case": null,
  "unit-no-unknown": Array [
    true,
  ],
  "value-list-comma-newline-after": null,
  "value-list-comma-newline-before": null,
  "value-list-comma-space-after": null,
  "value-list-comma-space-before": null,
  "value-list-max-empty-lines": null,
  "value-no-vendor-prefix": Array [
    true,
  ],
}
`;
